### ISSUE TITLE
Android TV/button navigation needs a rethink/redesign

### ISSUE THREAD

**Issue by opusforlife2** [TeamNewPipe/NewPipe#11197]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I'm aware that this is a request for NewPipe itself and that requests for adding a new service need to be made at [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/issues).
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Feature description

Newpipe should be completely navigable and accessible using Android TV or similar devices. There should be no touchscreen-exclusive actions.

### Why do you want this feature?

Plenty of issues opened about TV navigation being problematic. I'm closing them all in favour of a single, easier-to-track issue. This will have to be done during/post rewrite anyway.

**Comment by dasassaa** [comment]
Please, also add #11194

**Comment by logos88** [comment]
It wouldn't be more simple to use [smarttube](https://github.com/yuliskov/smarttube) as a TV player through DIAL protocol like [iBicha](https://github.com/iBicha) proposes here?:
https://github.com/TeamNewPipe/NewPipe/issues/10791#issuecomment-1937067692

This way NewPipe efforts can go to continue improving the app on phone and integrating it with other apps from other platforms

**Comment by jeaye** [comment]
What an amazing project! I just installed NewPipe last night on LineageOS on an rpi5, hoping to be able to use a PS5 controller to browse YouTube. To be honest, I was surprised how well it works, but there are some buttons/features I can't access with just the directional inputs. Namely, I noticed:

* The search icon in the top right
* The full screen button in the video player

For both of these, I can use the controller's touch pad to simulate mouse input, but I'd really love if this use case were considered for your rewrite. Thanks so much for all the hard work. :bow:

**Comment by jeaye** [comment]
> > What an amazing project! I just installed NewPipe last night on LineageOS on an rpi5, hoping to be able to use a PS5 controller to browse YouTube. To be honest, I was surprised how well it works, but there are some buttons/features I can't access with just the directional inputs. Namely, I noticed:
> > 
> > * The search icon in the top right
> > * The full screen button in the video player
> > 
> > For both of these, I can use the controller's touch pad to simulate mouse input, but I'd really love if this use case were considered for your rewrite. Thanks so much for all the hard work. 🙇
> 
> Hey, how did you manage to get to button to reopen player once its minimised?

I use the PS5 mouse pad to do any clicking necessary, when directional inputs are insufficient.

**Comment by celestguy** [comment]
Hey guys ! I'm up for giving it a try !

**Comment by peat80** [comment]
That would be awesome if you could improve Newpipe's TV UI. 😍

**Comment by lydenyardley** [comment]
@celestguy incredible!!! Anything you can fix would he SO appreciated 😊

**Comment by celestguy** [comment]
Hi folks, thanks for your enthusiasm! 
Right now, I’m both diving into the codebase to get a solid grasp of how everything works, and researching TV UI design best practices with Jetpack Compose. The goal is to create a smooth, lightweight, and easily maintainable interface for Android TV.

Please, feel free to ask any questions, or share any of your ideas !

**Comment by lydenyardley** [comment]
I would have thought it can stay as-is mainly to keep codebase consistent with regular android. In my mind, it needs some key fixes (like when on a channel page, you can't navigate down to click on any of the actual videos... And not being able to reorder playlists by long-pressing on one and getting a move up or move down option).

Do you think phase 1 work could be fixing basic functionality within the context that already exists?

**Comment by celestguy** [comment]
You’re absolutely right: phase 1 should focus on modernizing the existing codebase by migrating to Jetpack Compose, as outlined in the “refactor” post on the NewPipe website.

In my opinion, trying to adapt the current UI for Android TV would be less effective than building a dedicated TV activity from scratch. The existing interface wasn’t designed for TV navigation, and retrofitting it would likely result in a poor user experience. Developing a new, TV-optimized interface should be treated as a standalone feature—one that aligns with the refactoring principles already established for the app, ensuring a clean, maintainable architecture.

Of course I am open to discussion.

**Comment by lydenyardley** [comment]
You're prob right. I've only done HTML and CSS, so unfortunately dont have any understanding of Android development. I'm glad to have you here, working to improve the project in the bigger picture!

**Comment by peat80** [comment]
A standalone dedicated tv ui would be probably way easier to develop and maintain as waisting time trying to make the normal ui  complete working as is.

For example, the miniplayer is essential, but in its current form totally useless for non touch devices.

We would need one totally different to work in tv.
Maybe something like vlc tv ui does?

But i think the tv ui should be developed with additional touch support i think. This would be useful for tablets.
So we would get a nice new ui for tv and tablets. 😀

**Comment by celestguy** [comment]
Hi !

Work still in progress, guys, i'll probably share you some screenshots in a few days. 

Théo.

**Comment by celestguy** [comment]
Hello guys. 

Well, I'm really struggling right now, because I have to rewrite the entire app using Jetpack Compose, on top of learning Android development (I'm a junior developer—maybe a little too ambitious). I wanted to share some screenshots with you, but at this stage, it doesn't look very good. 

The NewPipe UI for Android TV will have to wait a little longer until the app itself is rewritten with Jetpack Compose...

However, it might be worth discussing here which features are actually useful for Android TV.
For example, I don’t think we should keep downloads, since TV storage space is pretty limited.

Feel free to let me know what you think.

Until then, I'm sorry for giving you false hope...
Théo.

**Comment by peat80** [comment]
Thats sad to hear, but understandable.
This is for sure no easy task.

And indeed tv has other usecases than downloading videos.

Also comments and such things i would not need for tv.
Pip is also useless for me on tv.
